lab-katy/model/user.js

```diff
@@ -0,0 +1,75 @@
+'use strict';
+
+const crypto = require('crypto');
+const bcrypt = require('bcrypt');
+const jwt = require('jsonwebtoken');
+const mongoose = require('mongoose');
+const createError = require('http-errors');
+const Promise = require('bluebird');
+const debug = require('debug')('cfgram:user');
+
+const Schema = mongoose.Schema;
+const userSchema = Schema({
+  username: { type: String, required: true, unique: true }, 
+  email: { type: String, required: true, unique: true },
+  password: { type: String, required: true },
+  findHash: { type: String, unique: true },
+});
+
+userSchema.methods.generatePasswordHash = function(password) {
+  debug('generatePasswordHash');
+
+  return new Promise((resolve, reject) => {
```

Comment:
I'm pretty sure that bcrypt returns promises, so you should be able to chain directly on